jung-algorithms/src/main/java/edu/uci/ics/jung/layout/event/LayoutChange.java

```diff
@@ -0,0 +1,86 @@
+package edu.uci.ics.jung.layout.event;
+
+import com.google.common.collect.Lists;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * For the most general change to a LayoutModel. There is no Event payload, only an indication that
+ * there was a change. A visualization would consume the event and re-draw itself. Use-cases for
+ * firing this event are when the Graph or LayoutAlgorithm is changed in the LayoutModel
+ *
+ * @author Tom Nelson
+ */
+public interface LayoutChange {
+
+  /** indicates support for this type of event dispatch */
+  interface Producer {
+    LayoutChange.Support getLayoutChangeSupport();
+  }
+
+  /** method signatures to add/remove listeners and fire events */
+  interface Support {
```

Comment:
There's a lot of structural and semantic similarity between the Support interfaces that you've defined.  Consider creating an abstract class that these could use; it would save a fair amount of redundant code and would let the individual support classes focus on the things that are unique to them.
(You could also do this as an interface, but the implementations are also structurally the same, so you may as well just do that once also.)
```java
public abstract class LayoutChangeListening<L> {
  private boolean enabled;
  // consider making this a Set so that we don't have to worry about duplicates
  private List<L> listeners = Collections.synchronizedList(Lists.newArrayList());
  boolean isEnabled() {  // or “enabled()”
    return enabled;
  }
  abstract void enable(boolean enable);
  void addListener(L listener) {
    listeners.add(listener);
  }
  void removeListener(L listener) {
    listeners.remove(listener);
  }
  List<L> listeners() {
    return listeners;
  }
}
```
and then you'd extend that inside these three interfaces:
```java
// inside LayoutChange
class Listening extends LayoutChangeListening<Listener> {
  public void enable(boolean enable) {
    this.enabled = enable;
    if (enabled) {
      fireChange();
    }
  }
  public void fireChange() {
    for (Listener listener : listeners()) {
      listener.layoutStateChanged();
    }    
  }
  public static Listening create() {
    return new Listening();
  }
}
// etc.
```
creating an instance: 
```java
LayoutStateChange.Listening = LayoutStateChange.Listening.create();
```

Reply:
I'll consider it. I separated them in order to make it really clear what the different event types are. I'll consider whether joining the similar parts helps me with that.

Reply:
The listener collections are Lists because that is the tradition. Implementing equals and hashCode would be a bit of overkill, too.
https://stackoverflow.com/questions/2938244/why-are-listener-lists-lists

Reply:
Set vs List: A Listener with a poorly implemented hashCode/equals could steal another listener's events, resulting in something extra hard to debug. Users write code that implements the Listeners.